core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1281,6 +1311,12 @@ private[spark] class DAGScheduler(
     stage match {
       case s: ShuffleMapStage =>
         outputCommitCoordinator.stageStart(stage = s.id, maxPartitionId = s.numPartitions - 1)
+        // Only generate merger location for a given shuffle map stage once. This way, even if
+        // this stage gets retried, it would still be merging blocks using the same set of
+        // shuffle services.
+        if (pushBasedShuffleEnabled && s.shuffleDep.shuffleMergeEnabled) {
```

Comment:
Is it necessary to check `pushBasedShuffleEnabled` here? If yes can add to the comment when will the first condition be false and the second one be true?

Reply:
This is not required, will remove this part.